### PR TITLE
Backbutton fix 7

### PR DESCRIPTION
**Pull request by kapapp**
Fixed app crashing with login/back button. Back button in main menu minimizes app (this is consistent with commercial app behavior), back button in NewUserInfo logs out and leads to login screen. Added kdoc and tests
!Added back deleted code!

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindwar/menu/MainMenuActivity.kt [hunk 1]
```diff
@@ -36,4 +42,12 @@ class MainMenuActivity : AppCompatActivity() {
             commit()
         }
     }
+
+    /**
+     * Minimizes the app if the back button is pressed
```
Comment (marluxiaboss) [inline app/src/main/java/ch/epfl/sdp/blindwar/menu/MainMenuActivity.kt:47]: Maybe it's not necessary to have this much space.

**Review by marluxiaboss — APPROVED** [review]
Good changes! But it would be better to create kdoc when actually creating the function, backtracking takes time.

**Review by ghemDD — APPROVED** [review]
Nice work 
Maybe add short description of parameters in doc when the name is not self-descriptive

File: app/src/main/java/ch/epfl/sdp/blindwar/login/UserNewInfoActivity.kt [hunk 2]
```diff
@@ -91,12 +94,21 @@ class UserNewInfoActivity : AppCompatActivity() {
...
-//        AuthUI.getInstance().signOut(this)
-//        AuthUI.getInstance().delete(this)
-//    }
+    /**
+     * To avoid crashing the app, if the back button is pressed, user will log out
+     *
+     */
+    override fun onBackPressed() {
+        //this.moveTaskToBack(true);
+        auth.signOut()
```
Comment (CHOCOLVT) [inline app/src/main/java/ch/epfl/sdp/blindwar/login/UserNewInfoActivity.kt:103]: I think that this should bu true only if user is a new one (and thus, we should also delete the newly created account on firebase)

Comment (kapapp) [reply app/src/main/java/ch/epfl/sdp/blindwar/login/UserNewInfoActivity.kt:103]: yeah, that's true, I'll do a separate PR for that

**Pull request merged**